src/main/java/seedu/address/logic/commands/UnlockCommand.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.logic.commands;
+
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.storage.Authentication;
+
+public class UnlockCommand extends Command {
+
+    public static final String COMMAND_WORD = "unlock";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Unlocks ClientBook.\n"
+            + "Parameters: CURRENT_PASSWORD\n"
+            + "Example: " + COMMAND_WORD + " 12345";
+
+    public static final String MESSAGE_UNLOCK_SUCCESS = "Unlocked ClientBook.";
+
+    public static final String MESSAGE_ALREADY_UNLOCKED = "ClientBook is not locked.";
+
+    public static final String MESSAGE_INCORRECT_PASSWORD = "You have entered the wrong password.";
+
+    private final String currentPassword;
+
+    /**
+     * Instantiates an UnlockCommand object with the password that the user has entered.
+     * @param currentPassword user entered password used to unlock the zip file.
+     */
+    public UnlockCommand(String currentPassword) {
+        this.currentPassword = currentPassword;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        Authentication authentication = model.getAuthentication();
+        if (!authentication.isPasswordPresent()) {
+            return new CommandResult(MESSAGE_ALREADY_UNLOCKED);
+        }
+        if (!authentication.getPassword().equals(currentPassword)) {
+            throw new CommandException(MESSAGE_INCORRECT_PASSWORD);
+        }
+        authentication.removePassword();
```

Comment:
If we remove the password here,  does that mean that every time the user unlocks the ClientBook, the next time he locks he needs to set a password again? 

If that's the case, then I'm thinking if it is possible that we don't remove password on unlock, so the user only needs to specify a password for the lock command when he/she first adds or changes his/her password. Then subsequently, if he/she locks and unlocks it will just use the same pasword.

Reply:
That is a great suggestion actually. 

One possible way would be to encrypt a password file using a secret key within our code which when the program runs, will be decrypted and stored in program state and encrypted immediately. 

The user can then enter lock without any parameters to lock using the old password.

However, I do feel that the effort to return ratio is rather minimal as it does not make our program any more tailored for our user than what already exists.

Reply:
I agree that the effort to reward ratio is pretty low since it's not a particularly core feature of our application. In that case, let's keep this extension in mind, and if we have time after other core features then we can possibly come back to it in the future.